share/wake/lib/system/remote_cache_api.wake

```diff
@@ -325,6 +334,27 @@ export def rscApiFindMatchingJob (req: CacheSearchRequest) (api: RemoteCacheApi)
 
     mkCacheSearchResponse json
 
+# rscApiCheckAuthorization: Checks if the provided authorization key is valid. 
+#
+# ```
+#  api | rscApiCheckAuthorization = Pass Unit
+# ```
+export def rscApiCheckAuthorization (api: RemoteCacheApi): Result Unit Error =
+    require Some auth = api.getRemoteCacheApiAuthorization
+    else failWithError "rsc: Checking authorization requires authorization but none was provided"
```

Comment:
I don't know what sort of logic chokepoints you have closer to the surface and with this being `export` it might indeed be the best way to handle it,[^1] but don't forget that if everything ultimately funnels through a small number of narrow necks, you can collect all the `rsc:` prefixes through `addErrorContext` and save a bit of typing/typo risk.

[^1]: Actually, why is it exported?  Doesn't seem like anything which would be helpful for general audiences, but I don't know what sort of package organization you're planning.

Reply:
> why is it exported

I consider any HTTP web route as a public part of the `RemoteCacheApi` interface. Anyone with an instance of the apii should be able to call any function on the remote server.

Since `POST auth/check` is a function made available on the server, anyone should be able to call it.

 > If everything ultimately funnels through a small number of narrow necks, you can collect all the rsc: prefixes
 
I'll need you to give an example of what you mean, but maybe given the above comment this doesn't make sense anymore? The important thing to keep in mind is that this library is just a wrapper around what is available as the public interface on the server. What the consumer does with it (in this case remote_cache_runner) is entirely up to them

Reply:
Yeah, part 1 makes sense and with that clarification, part 2 is indeed invalid.  I basically just meant that if you had a lot of potential errors which were not in directly-exported functions but instead were only accessible by going through a small number of entry points, you wouldn't have to add the `rsc:` prefix to each of the errors at the place you throw them, and could instead call `addErrorContext "rsc"` at those chokepoints, for what would likely be fewer times than you have errors.